### ISSUE TITLE
Why not implement this as a directive ?

### ISSUE THREAD

**Issue by lionel-bijaoui** [FranckFreiburger/vue-resize-sensor#3]
Since all this does is add a new type of event, I think it is better suited to be a directive (v-resize for example) instead of a component. That way, you can attach it to any element in the page with less visual clutter.

```html
<div v-resize @resize="resize"><div>
```

At first, I though it would be impossible since you need the additional elements (`<div @scroll="scroll"><div></div></div><div @scroll="scroll"><div></div></div>`) but [this script ](https://github.com/marcj/css-element-queries/blob/master/src/ResizeSensor.js) seem to work on any element and create the additional element needed without modifying/breaking the original element.

Why this choice ? Am I missing something ?
Thank you.

**Comment by FranckFreiburger** [comment]
I agree that it could be very convenient to have a directive. `resize-sensor` needs to add an element inside the container you are watching for resizing. This mean that all element are not suitable for this operation.
For example, <img v-resize> will fail.

**Comment by lionel-bijaoui** [comment]
Well, maybe add a warning for the edge cases ? They could still do something like that:
```html
<div>
    <div v-resize @resize="resize"><div>
    <img>
</div>
```
Well, that was just my two cents, good luck for the future 👍
